File: android/features/src/main/kotlin/dev/example/moodlenative/features/HomeRemainingCourse.kt
package dev.example.moodlenative.features

data class HomeRemainingCourse(
    val course: LmsCourseSummary,
    val periodIndex: Int,
    val room: String?,
) {
    val id: String
        get() = "${course.id}-$periodIndex"

    val periodTitle: String
        get() = "${periodIndex + 1}限"
}

File: ios/Shared/Sources/Features/HomeRemainingCourse.swift
import Foundation
import MoodleNativeCore

public struct HomeRemainingCourse: Identifiable, Equatable, Sendable {
    public let course: LmsCourseSummary
    public let periodIndex: Int
    public let room: String?

    public init(course: LmsCourseSummary, periodIndex: Int, room: String?) {
        self.course = course
        self.periodIndex = periodIndex
        self.room = room
    }

    public var id: String {
        "\(course.id)-\(periodIndex)"
    }

    public var periodTitle: String {
        "\(periodIndex + 1)限"
    }
}
